packages/daimo-expo-enclave/android/src/main/java/expo/modules/enclave/FallbackKeyManager.kt

```diff
@@ -81,4 +112,26 @@ class FallbackKeyManager: KeyManager {
     }
     return verified
   }
+
+  override fun onActivityResult(activity: Activity, requestCode: Int, resultCode: Int, data: Intent?) {
+    if (requestCode == DEVICE_CREDENTIAL_FALLBACK_CODE) {
```

Comment:
just to check, we're sure that no other activity is using code 1001  = onActivityResult will always be called exactly once per `sign()`?

Reply:
may be simplified slightly to

```kotlin
if (requestCode != DEVICE_CREDENTIAL_FALLBACK_CODE)  return;
if (pendingSign == null) return;
try {
  if (resultCode == Activity.RESULT_OK) {
    // ...
    pendingSign.promise.resolve(signature.toHexString());
  } else {
    pendingSign.promise.reject("ERR_AUTH_FAILED", "Authentication failed")
  }
} finally {
  pendingSign = null;
}
```